### PR TITLE
Modernize dependencies, HTTP and JSON libraries

### PR DESCRIPTION
**Pull request by jamesvl**
`Jason` has been the preferred JSON library in Elixir for some time. `HTTPoison` works but newer libraries like Mint, Finch, and `Req` have sprung up as more idiomatic (and more secure by default) to Elixir.

This updates all libraries and replaces HTTPoison and Poison with Req and Jason.

* https://hexdocs.pm/req/readme.html
* https://hexdocs.pm/jason/readme.html

### CODE REVIEW

File: lib/vaultex/redirectable_requests.ex [hunk 1]
```diff
@@ -1,56 +1,43 @@
 defmodule Vaultex.RedirectableRequests do
```
Comment (guessthepw) [inline lib/vaultex/redirectable_requests.ex:1]: We don't redirect here anymore at all, should this module be renamed?

Comment (jamesvl) [reply lib/vaultex/redirectable_requests.ex:1]: I kept it mainly for legacy reasons; but Req by default *does* follow redirects, so technically the name is still accurate (if a bit wordy).

**Review by guessthepw — APPROVED** [review]
